### ISSUE TITLE
Aliases of InitVar not recognized

### ISSUE THREAD

**Issue by Aran-Fey** [microsoft/pyright#6364]
**Describe the bug**
`dataclasses.InitVar` isn't recognized if it's stored in a variable. (Why do I want to do that? Because I'm writing my own dataclass-like library.)

```python
import dataclasses

InitVar = dataclasses.InitVar

@dataclasses.dataclass
class Foo:
    foo: InitVar[int]

    def __post_init__(self, foo: int):
        pass
```

Output:

> Dataclass `__post_init__` incorrect parameter count; number of InitVar fields is 0

**VS Code extension or command-line**
Both. The command-line version is 1.1.334.

**Comment by erictraut** [comment]
Yes, this is by design. `InitVar` is a special form that requires special processing early in the type checking process prior to when expression types can be evaluated. If you want your code to work with pyright, you'll need to directly use the `InitiVar` symbol without aliasing it.

**Comment by Aran-Fey** [comment]
I really need to stop wasting my time writing bug reports.

**Comment by erictraut** [comment]
Pyright does contain special-case handling that enables local aliasing of `InitVar` using the `as` form of an `import` statement. Perhaps this is sufficient for your use case?

```python
from dataclasses import InitVar as LocalInitVar, dataclass

@dataclass
class Foo:
    foo: LocalInitVar[int]

    def __post_init__(self, foo: int):
        pass
```

Pyright doesn't support aliasing _and_ re-exporting of the following special forms: `InitVar`, `ClassVar`, `Final`, or `Annotated`. These all require processing in the early phases of analysis (during the "binding" stage, which identifies scopes and builds a symbol table for each scope).

**Comment by Aran-Fey** [comment]
Sadly that doesn't cut it for my use case. I'd like my library to re-export it so that it's available as `my_library.InitVar`.
